pkg/restapi/vc/operation/operations.go

```diff
@@ -828,39 +830,145 @@ func (o *Operation) issueCredentialHandler(rw http.ResponseWriter, req *http.Req
 		return
 	}
 
-	// Resolve DID and get the public keyID
-	didDoc, err := o.vdri.Resolve(cred.Opts.AssertionMethod)
+	// sign the credential
+	signedVC, err := o.signCredential(validatedCred, cred.Opts.AssertionMethod)
 	if err != nil {
-		o.writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf("failed to resolve DID : %s", err.Error()))
+		o.writeErrorResponse(rw, http.StatusInternalServerError, fmt.Sprintf("failed to sign credential:"+
+			" %s", err.Error()))
 
 		return
 	}
 
-	publicKeyID, err := getPublicKeyID(didDoc)
+	rw.WriteHeader(http.StatusOK)
+	o.writeResponse(rw, signedVC)
+}
+
+func (o *Operation) composeAndIssueCredentialHandler(rw http.ResponseWriter, req *http.Request) {
+	// get the request
+	composeCredReq := ComposeCredentialRequest{}
+
+	err := json.NewDecoder(req.Body).Decode(&composeCredReq)
```

Comment:
Any followups for the two query parameters, `templateReference` and `subjectReference`?

Reply:
https://github.com/trustbloc/edge-service/issues/144